Tdi sde wrapper

ffoulkes opened the pull request.

ffoulkes committed:
Modularize tdi_sde_wrapper

- Partitioned tdi_sde_wrapper.cc into multiple source files.

- Renamed GetBfChipType() method to GetChipType().

Signed-off-by: Derek G Foster <derek.foster@intel.com>

ffoulkes committed:
Minor SDE wrapper edits

- Also addressed clang compiler errors in tofino_switch.h.

Signed-off-by: Derek G Foster <derek.foster@intel.com>

ffoulkes committed:
Update Bazel build for modularized tdi_sde_wrapper

Signed-off-by: Derek G Foster <derek.foster@intel.com>

ffoulkes committed:
Restore Bazel build dependency that was accidentally deleted

- Restore defines=target_defines in //stratum/hal/lib/tdi:macros.
  The macros.h file imports bf_types/bf_types.h.

Signed-off-by: Derek G Foster <derek.foster@intel.com>

ffoulkes commented on stratum/hal/lib/tdi/BUILD:
```diff
@@ -278,7 +299,6 @@ stratum_cc_library(
 stratum_cc_library(
     name = "macros",
     hdrs = ["macros.h"],
-    defines = target_defines,
```
Accidental deletion; fixed in a subsequent change.
`macros.h` #includes `bf_types.h` from the SDE.

ravi861 commented on stratum/hal/lib/tdi/tdi_sde_wrapper.h:
```diff
@@ -25,6 +23,10 @@
 #include "stratum/hal/lib/common/common.pb.h"
 #include "stratum/lib/channel/channel.h"
 
+#ifdef TOFINO_TARGET
```
Can we move this into tdi_sde_wrapper.cc? I am not sure why it is even here in the first place.

ffoulkes replied:
No, we can't. The #ifdef is here because the %$@#! Packet I/O methods were defined using Tofino-specific symbols. They should be replaced with abstract interfaces that are SDE-neutral.

This is why I asked about the Packet I/O interface earlier today. I really want the TDI team to fix this.

ravi861 commented on stratum/hal/lib/tdi/tdi_sde_wrapper.cc:
```diff
@@ -3336,302 +374,422 @@ ::util::Status TdiSdeWrapper::GetActionProfileGroups(
 }
 
-::util::Status TdiSdeWrapper::ResetDefaultTableEntry(
+::util::Status TdiSdeWrapper::ReadIndirectMeters(
```
Meters are also special table types like action profiles. They can also probably get their own files.

ffoulkes replied:
True. Meters, Counters, Registers, and CloneSessions are all candidates for additional refactoring.

ravi861 reviewed (approved):
Does the keyword sde make any sense anymore? Maybe just rename all tdi_sde_* files to tdi_*?

ffoulkes committed:
Continue refactoring tdi_sde_wrapper

- Created another four source files from tdi_sde_wrapper.cc.

- Updated CMake and Bazel build files to reflect the change.

Signed-off-by: Derek G Foster <derek.foster@intel.com>
